x-pack/packages/kbn-ai-assistant/src/conversation/conversation_view.tsx

```diff
@@ -61,10 +61,10 @@ export const ConversationView: React.FC<ConversationViewProps> = ({
   );
 
   useEffect(() => {
-    if (scope) {
-      service.setScope(scope);
+    if (scopes) {
+      service.setScopes(scopes);
     }
-  }, [scope, service]);
+  }, [scopes, service]);
```

Comment:
can we use an isEqual check here? it should be fairly cheap and then we don't have to worry about people not memoizing `scopes` upstream.

Reply:
Fixed in https://github.com/elastic/kibana/pull/196322/commits/2e2bdb9941e1215ec321e1fbff9b41b87b38c0b1